### ISSUE TITLE
Getting csa lib to work on Mac

### ISSUE THREAD

**Issue by drf5n** [hetland/pygridgen#29]
Per https://github.com/hetland/pygridgen/issues/21#issuecomment-359995326 it looks like the old solution for OSX is to get octant to compile, then use those libraries.

Lots of the hints point towards older googlecode externals, like for csa-c but those have broken URLs.

I can get a fresher csa-c link from https://github.com/sakov/csa-c.git  per https://github.com/hetland/octant/issues/4#issuecomment-360003688 and it seems to compile and install some shared libraries with this bit of code:
```

git clone https://github.com/sakov/csa-c.git
pushd csa-c/csa
./configure --prefix=$CONDA_PREFIX # or whereever
make clean install
popd

```

But I still get an error around https://github.com/hetland/pygridgen/blob/master/pygridgen/csa.py#L62 where:

    path = os.path.join(site.getsitepackages()[0], 'pygridgen')
    _csa = np.ctypeslib.load_library('_csa', path)

I made a couple adjustments in my extern/csa/makefile to create an OSX libcsa.dylib file and adjusted the csa.py code to read like:

```
    path = os.path.join(os.getenv('CONDA_PREFIX'), 'lib')
    try:
       _csa = np.ctypeslib.load_library('libcsa', path)
    except:
       print "Couldn't find libcsa in {}".format(path)
       raise

```

A bit of my csa/makefile looks like: 

```
#SHLIB = libcsa.so
#SHLIBFLAGS = -shared -fPIC

SHLIB = libcsa.dylib
SHLIBFLAGS = -dynamiclib -fPIC

all: libcsa.a $(SHLIB) $(PROGRAMS)

...
$(SHLIB): $(LIBOBJS)
        rm -f $(SHLIB)
        $(CC) $(SHLIBFLAGS)  -o $@ $(LIBOBJS)
        chmod go+r $(SHLIB)

...

```

... then things worked as expected.

To make csa compile on OSX automatically, you'd need to adjust the extern/makefile.in to determine OSX or Linux and do the different dynamic library setups.  Sounds more like an issue for csa-c

**Comment by phobson** [comment]
The conda-forge recipes for pygridgen and its dependencies have been very reliable on Mac OS and linux for a while now. For CSA in particular, the recipe is pretty simple:
https://github.com/conda-forge/csa-feedstock/blob/master/recipe/build.sh

My recommendation is to get the dependencies from conda-forge.

**Comment by drf5n** [comment]
Are you saying a Mac install works for you if you do something like: 

    conda config --add channels conda-forge
    
    conda install csa nn

and do something else for gridutil and gridgen? 


Ack -- I guess this is the fork to try rather than Rob's:

* https://phobson.github.io/pygridgen/install/build_osx.html

Rob's instructions on https://github.com/hetland/pygridgen/blob/master/doc/INSTALL_OSX.md#cloning-repos-from-github has a bad link, and it looks like https://phobson.github.io/pygridgen/install/build_osx.html#cloning-dependencies-from-github lists the gihub repositories explicitly.

**Comment by phobson** [comment]
That's my documentation page. It's out of date and I need to update it.

But yes:

`conda create name=grid python=3.6 pygridgen --channel=conda-forge`

will install pygridgen and all of the dependencies (including csa, nn, gridgen, and gridutils) on linux and Mac OS.

We still don't have Windows build ready, though the above command works within the latest Windows Subsystem for Linux (WSL). Now that jupyter notebook servers spawned in WSL can take to a native browser in Windows, my motivation for figuring out how to build natively on Windows has dropped dramatically.

**Comment by drf5n** [comment]
My conda didn't like the `name=` syntax.  maybe:

    conda create grid python=3.6 pygridgen --channel=conda-forge


Looking at https://phobson.github.io/pygridgen/install/build_osx.html#cloning-dependencies-from-github you could do the first two with conda and then put gridgutils and gridgen in with the $CONDA_PREFIX

**Comment by phobson** [comment]
@drf5n 
> My conda didn't like the name= syntax. maybe:

Yeah I messed that up. It's edited now. Should have been `--name=grid`

**Comment by cvdwl** [comment]
Might be worthwhile updating the README to mention the relative maturity of the conda-forge package; we just spent several days in the same csa rabbit-hole before I found my way into the conda-forge solution via a comment of Rich's.

**Comment by phobson** [comment]
@cvdwl I'll soon be taking the lead at cleaning up and documenting these tools. 

Rich has created a new github organization for the "ecosystem" https://github.com/pygridgen

Watch that space for more info, and please do submit a PR once the repositories are transferred over.

**Comment by drf5n** [comment]
That space seems broken.

It might be nice to have a new centralized place, but the issue I have is that it is so hard to find one's way to the newer & better places/toolchains from the older tools and sources.

Take a look at https://www.myroms.org/wiki/Grid_Generation as an basic entry point into the ecosystem.

**Comment by phobson** [comment]
What do you mean by broken?

**Comment by drf5n** [comment]
I mean it has nothing visible in it: "This organization has no public repositories"

**Comment by phobson** [comment]
@drf5n  that's to be expected. Like I said, "I'll *soon* take the lead...".

Things will be fleshed out at some point in the near future, once my funding and scope are fully administered.

I don't think centralizing the toolchain and its documentation on the ROMS wiki is a good idea since the tools are applicable to much more than ROMS. Personally, I've never even seen a ROMS model. I use EFDC, CE-QUALW2, and GEMS.

**Comment by drf5n** [comment]
OK, but it is hard to watch/subscribe when there's nothing there.  Why not fork to a new place and edit?

I didn't mean to recommend ROM's wiki as the best place.  I meant that as an example of transferring things to new places breaking links. Several in the labs I work with are using ROMS, and it's recommendations of grid generation software are aging and broken.  There probably ought to be a link to pygridgen there, but that's an issue to take up with the ROMS wiki maintainers.  As to what URL should be put there, https://github.com/pygridgen isn't yet ready, https://github.com/hetland/pygridgen seems out of date as compared to your fork at https://github.com/phobson/pygridgen . What link would you recommend for their wiki?

Whichever it should be, the READMEs for the existing parent and intervening repositories should point folks towards the current or more recent packages.

**Comment by phobson** [comment]
@drf5n that'll be the point of the `pygridgen` organization. we're working on it.
